vector/src/debug/java/im/vector/app/features/debug/features/DebugVectorOverrides.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2022 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.debug.features
+
+import android.content.Context
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.booleanPreferencesKey
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.preferencesDataStore
+import im.vector.app.features.VectorOverrides
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import org.matrix.android.sdk.api.extensions.orFalse
+
+private val Context.dataStore: DataStore<Preferences> by preferencesDataStore(name = "vector_overrides")
+private val forceDialPadDisplay = booleanPreferencesKey("force_dial_pad_display")
+private val forceLoginFallback = booleanPreferencesKey("force_login_fallback")
+
+class DebugVectorOverrides(private val context: Context) : VectorOverrides {
+
+    override fun forceDialPad() = forceDialPadDisplayFlow
+    override fun forceLoginFallback() = forceLoginFallbackFlow
+
+    suspend fun setForceDialPadDisplay(force: Boolean) {
+        context.dataStore.edit { settings ->
+            settings[forceDialPadDisplay] = force
+        }
+    }
+
+    private val forceDialPadDisplayFlow: Flow<Boolean> = context.dataStore.data.map { preferences ->
+        preferences[forceDialPadDisplay].orFalse()
+    }
```

Comment:
Is there any reason we have these separate from the two override functions above? I think it would be much simpler to remove these vals and move their bodies to those functions

Reply:
the 2 members of the interface could also be `val`s I think.

Reply:
will inline into vals :+1:

Reply:
updated https://github.com/vector-im/element-android/pull/5361/commits/7637ee45447f052cc1675dd7bb1c589c165a2659